index.js

```diff
@@ -113,6 +113,11 @@ class PQueue {
 	}
 
 	onEmpty() {
+		// Instantly resolve if the queue is empty
+		if (this.queue.size === 0) {
```

Comment:
Here should be `&& this.pending === 0`, fixed this in master.

Reply:
Or should it be?

Reply:
Ah, cool. Wasn't sure if you guys were okay with this already... Should this also be fixed in `this._resolveEmpty`?

Reply:
I see two options:

1. Promise from onEmpty resolves when `size === 0 && pending === 0`
2. Two separate methods; onEmpty and onIdle (not sure about naming, but you get the gist)

Reply:
@edorivai I'm not sure, but is there a case, when we only need `size === 0` for queue?

@sindresorhus It was long time ago, but is this behaviour of `.onEmpty` intentional?

If we are about to bump major version, this should be reconsidered.

Reply:
I would go for second option, because it matches with [async.queue](https://github.com/async-js/async#queue) naming.

Reply:
But from `Advanced example` it looks like first option is anticipated – and I guess this is what most people expect from queue. If we add some work to it – `onEmpty` should mean, that all work is done.

Reply:
Shall we open a separate issue for the `pending === 0` functionality?

Reply:
@edorivai sure.